### PR TITLE
Create ETL task to map columns

### PR DESCRIPTION
**Pull request by Yun-Wu**
- Add a column mapping task
- Add the task to registry
- Add unittest

https://github.com/wildlife-dynamics/ecoscope-workflows/issues/34

### CODE REVIEW

**Comment by Yun-Wu** [comment]
@cisaacstern thanks for the prompt review haha! Can you take a look again?

File: ecoscope_workflows/tasks/transformation/_column_mapping.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    df: DataFrame[JsonSerializableDataFrameModel],
+    drop_columns: Annotated[
+        list[str], Field(default=[], description="List of columns to drop.")
+    ],
+    retain_columns: Annotated[
+        list[str],
+        Field(
+            default=[],
+            description="""List of columns to retain with the order specified by the list.
+                        "Keep all the columns is the list is empty.""",
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/transformation/_column_mapping.py:20]:
```suggestion
                        "Keep all the columns if the list is empty.""",
```

File: ecoscope_workflows/tasks/transformation/_column_mapping.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        DataFrame[JsonSerializableDataFrameModel]: The transformed DataFrame.
+
+    Raises:
+        KeyError: If any of the columns specified are not found in the DataFrame.
+    """
+
+    df = df.drop(columns=drop_columns)
+    if retain_columns:
+        if any(col not in df.columns for col in retain_columns):
+            raise KeyError(f"Columns {retain_columns} not found in DataFrame.")
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/transformation/_column_mapping.py:48]:
```suggestion
            raise KeyError(f"Columns {retain_columns} not all found in DataFrame.")
```

File: ecoscope_workflows/tasks/transformation/_column_mapping.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        df = df.reindex(columns=retain_columns)
+    if rename_columns:
+        if any(col not in df.columns for col in rename_columns):
+            raise KeyError(f"Columns {retain_columns} not found in DataFrame.")
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/transformation/_column_mapping.py:52]:
```suggestion
            raise KeyError(f"Columns {rename_columns} not all found in DataFrame.")
```

**Review by cisaacstern — APPROVED** [review]
LGTM!

Commented on a few small nits.

The only thing from the linked issue I don't see explicitly covered here is:

> "The function should provide a warning when critical columns (e.g., 'geometry') are being dropped."

Not sure if you'd like to address that now or in a follow-on. Approving this to indicate that either is fine for me.

In general, I really like the pattern of a full module of tests in for a single task and all its edge cases, that focuses on the behavior of the function in "pure python" mode (without any of the distributed behaviors interfering with the clarity of the logic). I think we should adopt this pattern for all of the task testing. The testing I have for the existing tasks is too opaque!

File: ecoscope_workflows/tasks/transformation/_column_mapping.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        )
+
+    df = df.drop(columns=drop_columns)
+    if retain_columns:
+        if any(col not in df.columns for col in retain_columns):
+            raise KeyError(f"Columns {retain_columns} not all found in DataFrame.")
+        df = df.reindex(columns=retain_columns)
+    if rename_columns:
+        if any(col not in df.columns for col in rename_columns):
+            raise KeyError(f"Columns {retain_columns} not all found in DataFrame.")
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/transformation/_column_mapping.py:64]:
```suggestion
            raise KeyError(f"Columns {rename_columns} not all found in DataFrame.")
```

**Review by cisaacstern — APPROVED** [review]
Awesome, thanks Yun!!

One last typo noted below. Otherwise, if you're happy here, I'm good to merge!

File: ecoscope_workflows/tasks/transformation/_column_mapping.py (hunk 4 above)
Comment (Yun-Wu) [reply ecoscope_workflows/tasks/transformation/_column_mapping.py]: Thanks for the detailed review!
